### PR TITLE
[SPARK-25630][TEST] Reduce test time of HadoopFsRelationTest

### PR DESCRIPTION
**Pull request by gengliangwang**
## What changes were proposed in this pull request?
There was 5 suites extends `HadoopFsRelationTest`,  for testing "orc"/"parquet"/"text"/"json" data sources. 
This PR refactor the base trait `HadoopFsRelationTest`:
1. Rename unnecessary loop for setting parquet conf
2. The test case `SPARK-8406: Avoids name collision while writing files` takes about 14 to 20 seconds. As now all the file format data source are using common code, for creating result files, we can test one data source(Parquet) only to reduce test time.

To run related 5 suites:
```
./build/sbt "hive/testOnly *HadoopFsRelationSuite"
```
The total test run time is reduced from 5 minutes 40 seconds to 3 minutes 50 seconds.

## How was this patch tested?

Unit test

### CODE REVIEW

File: sql/hive/src/test/scala/org/apache/spark/sql/sources/HadoopFsRelationTest.scala [hunk 1]
```diff
@@ -760,23 +775,27 @@ abstract class HadoopFsRelationTest extends QueryTest with SQLTestUtils with Tes
   // requirement.  We probably want to move this test case to spark-integration-tests or spark-perf
   // later.
   test("SPARK-8406: Avoids name collision while writing files") {
```
Comment (gatorsmile) [inline sql/hive/src/test/scala/org/apache/spark/sql/sources/HadoopFsRelationTest.scala]: Just move this to ParquetHadoopFsRelationSuite.scala

Comment (dongjoon-hyun) [reply sql/hive/src/test/scala/org/apache/spark/sql/sources/HadoopFsRelationTest.scala]: +1

**Comment by gengliangwang** [comment]
retest this please.

**Comment by gengliangwang** [comment]
@dongjoon-hyun please take another look, thanks!

**Comment by gatorsmile** [comment]
LGTM

Thanks! Merged to master.

**Pull request closed without merging**